dotnet/src/SemanticKernel/SkillDefinition/SKFunction.cs

```diff
@@ -367,13 +367,16 @@ private struct MethodDetails
         public bool IsSensitive { get; set; }
     }
 
-    private static async Task<string> GetCompletionsResultContentAsync(IReadOnlyList<ITextCompletionResult> completions, CancellationToken cancellationToken = default)
+    private static async Task<string> GetCompletionsResultContentAsync(IReadOnlyList<ITextResult> completions, CancellationToken cancellationToken = default)
     {
         StringBuilder completionResult = new();
 
-        foreach (ITextCompletionResult result in completions)
+        foreach (ITextResult result in completions)
         {
             completionResult.Append(await result.GetCompletionAsync(cancellationToken).ConfigureAwait(false));
+
+            // To avoid any unexpected behavior we only take the first completion result (when running from the Kernel)
```

Comment:
Why have a List arg, if we are only returning the contents of the first item? I'd say the method should be more clear -- `Task<string> GetCompletion*ResultContentAsync(ITextResult completion, ...)`
or
`Task<IReadOnlyList<string>> GetCompletionsResultContentAsync(IReadOnlyList<ITextResult> completions`

Reply:
This method initially was combining the multiple results in one and I removed this behavior and instead of removing the whole method I kept it because in near term will be touching this need to return multiple results in the same prompt.

But you are right here, instead of keeping this method I can just remove it and rewrite it again in a later time when we fully support multiple responses as part of RunAsync impl.